apps/mobile/src/state/chat/atoms/checkAndDeleteOldChatsDataActionAtom.ts

```diff
@@ -0,0 +1,42 @@
+import {type PublicKeyPemBase64} from '@vexl-next/cryptography/src/KeyHolder'
+import {atom} from 'jotai'
+import {focusAtom} from 'jotai-optics'
+import {DateTime} from 'luxon'
+import removeFile from '../../../utils/removeFile'
+import messagingStateAtom from './messagingStateAtom'
+
+const ONE_MONTH_IN_MS = 1000 * 60 * 60 * 24 * 30
+
+export const checkAndDeleteOldChatsDataActionAtom = atom(
+  null,
+  (_, set, {key}: {key: PublicKeyPemBase64}) => {
+    const inbox = focusAtom(messagingStateAtom, (optic) =>
+      optic.find((one) => one.inbox.privateKey.publicKeyPemBase64 === key)
+    )
+
+    set(inbox, (prev) => ({
+      ...prev,
+      chats: prev.chats.filter((oneChat) => {
+        const isChatDeletedAndExpired = oneChat.messages.some(
+          (message) =>
+            (message.message.messageType === 'DELETE_CHAT' ||
+              message.message.messageType === 'INBOX_DELETED' ||
+              message.message.messageType === 'CANCEL_REQUEST_MESSAGING' ||
+              message.message.messageType === 'BLOCK_CHAT' ||
+              message.message.messageType === 'DISAPPROVE_MESSAGING' ||
+              message.message.messageType === 'OFFER_DELETED') &&
```

Comment:
Why not use already existing function for this. Such as: 

https://github.com/vexl-it/vexl/blob/c388e104d7feaec2dc3eddc28d0f90f92a625bf4/apps/mobile/src/state/chat/utils/isChatActive.ts

Reply:
The problem is, that INBOX_DELETED might be in the chat sent by the other side, in that case, we don't want to delete the chat before user deletes it first by clicking on "delete chat" button